crypto/bio/bss_dgram.c

```diff
@@ -577,14 +562,15 @@ static long dgram_ctrl(BIO *b, int cmd, long num, void *ptr)
         BIO_ADDR_make(&data->peer, BIO_ADDR_sockaddr((BIO_ADDR *)ptr));
         break;
     case BIO_CTRL_DGRAM_SET_NEXT_TIMEOUT:
-        memcpy(&(data->next_timeout), ptr, sizeof(struct timeval));
+        data->next_timeout = ossl_time_from_timeval(*(struct timeval *)ptr);
         break;
 # if defined(SO_RCVTIMEO)
     case BIO_CTRL_DGRAM_SET_RECV_TIMEOUT:
 #  ifdef OPENSSL_SYS_WINDOWS
         {
             struct timeval *tv = (struct timeval *)ptr;
             int timeout = tv->tv_sec * 1000 + tv->tv_usec / 1000;
```

Comment:
```
int timeout = (int)ossl_time2ms(ossl_time_from_timeval((struct timeval *)ptr));

Reply:
I thought it odd to convert to a time and then back to ms, when the direct conversion to ms was possible and fairly clear.  It also avoids the question of is the timeval a duration or a time?

Reply:
Possibly it's clear, I think it's a bit clearer with the name than magic number calculations.

Reply:
Still don't like the "magic" numbers...